### PR TITLE
dev-lang/zig: update to LLVM 14

### PR DESCRIPTION
**Pull request by BratishkaErik**
https://github.com/ziglang/zig/pull/12001
Signed-off-by: Eric Joldasov <bratishkaerik@getgoogleoff.me>

### CODE REVIEW

File: dev-lang/zig/zig-9999.ebuild [hunk 1]
```diff
@@ -23,22 +23,17 @@ RESTRICT="!test? ( test )"
...
 BUILD_DIR="${S}/build"
 
-# According to zig's author, zig builds that do not support all targets are not
-# supported by the upstream project.
-ALL_LLVM_TARGETS=(
-	AArch64 AMDGPU ARM AVR BPF Hexagon Lanai Mips MSP430 NVPTX
-	PowerPC RISCV Sparc SystemZ WebAssembly X86 XCore
-)
-ALL_LLVM_TARGETS=( "${ALL_LLVM_TARGETS[@]/#/llvm_targets_}" )
-LLVM_TARGET_USEDEPS="${ALL_LLVM_TARGETS[@]}"
```
Comment (BratishkaErik) [inline dev-lang/zig/zig-9999.ebuild:33]: We don't need to repeat this:

https://github.com/gentoo/gentoo/blob/987aa47ed905da959213f9fa8ef95da25c00e8c7/profiles/base/package.use.force#L34-L40

File: dev-lang/zig/zig-9999.ebuild [hunk 2]
```diff
@@ -23,22 +23,17 @@ RESTRICT="!test? ( test )"
...
 	sys-devel/clang:${LLVM_MAX_SLOT}
 	>=sys-devel/lld-${LLVM_MAX_SLOT}
 	<sys-devel/lld-$((${LLVM_MAX_SLOT} + 1))
-	sys-devel/llvm:${LLVM_MAX_SLOT}[${LLVM_TARGET_USEDEPS// /,}]
+	sys-devel/llvm:${LLVM_MAX_SLOT}
+"
+DEPEND="
+	${RDEPEND}
+	dev-util/cmake
+	>=sys-libs/zlib-1.2.12
```
Comment (thesamesam) [inline dev-lang/zig/zig-9999.ebuild:35]: Why would this only be in DEPEND?

File: dev-lang/zig/zig-9999.ebuild [hunk 3]
```diff
@@ -23,22 +23,17 @@ RESTRICT="!test? ( test )"
...
 RDEPEND="
 	sys-devel/clang:${LLVM_MAX_SLOT}
 	>=sys-devel/lld-${LLVM_MAX_SLOT}
 	<sys-devel/lld-$((${LLVM_MAX_SLOT} + 1))
-	sys-devel/llvm:${LLVM_MAX_SLOT}[${LLVM_TARGET_USEDEPS// /,}]
+	sys-devel/llvm:${LLVM_MAX_SLOT}
+"
+DEPEND="
+	${RDEPEND}
+	dev-util/cmake
```
Comment (thesamesam) [inline dev-lang/zig/zig-9999.ebuild:34]: This would be BDEPEND, if anything, but hte eclass does it for us. So drop it.

Comment (BratishkaErik) [reply dev-lang/zig/zig-9999.ebuild:34]: > This would be BDEPEND, if anything, but hte eclass does it for us. So drop it.

Thank you!

**Pull request closed without merging**